### PR TITLE
React: Fix entity enum and relationship dialog

### PR DESCRIPTION
**Pull request by agaspardcilia**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs [hunk 1]
```diff
@@ -139,7 +139,15 @@ export class <%= entityReactName %>Dialog extends React.Component<I<%= entityRea
...
-
+  <%_ if (hasRelationshipQuery) { _%>
+  clearEmptyRelations = entity => {
+    Object.keys(entity).filter(k => entity[k] instanceof Object).forEach(k => {
+      if (entity[k]['id'] === '') {
+        delete entity[k];
+      }
+    });
+  }
+  <%_ } _%>
```
Comment (agaspardcilia) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: Not the best way to do it, but it's pretty quick and does not add bloat to the code.

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs [hunk 2]
```diff
@@ -139,7 +139,15 @@ export class <%= entityReactName %>Dialog extends React.Component<I<%= entityRea
     this.props.<%= val.action %>();
     <%_ }) _%>
   }
-
+  <%_ if (hasRelationshipQuery) { _%>
+  clearEmptyRelations = entity => {
+    Object.keys(entity).filter(k => entity[k] instanceof Object).forEach(k => {
+      if (entity[k]['id'] === '') {
+        delete entity[k];
```
Comment (deepu105) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: React folks will kill us if they see use of delete, so no, lets not do this. Mutating stuff in React is heavily frowned upon

Comment (deepu105) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: could you explain the issue to me, maybe there is another way

Comment (agaspardcilia) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: When you don't put any value in a relationship field in a form, it creates a field in values that looks like this: `values.<relationship>.id: ""`. This by itself is no big deal but once you send that to the server it returns a 500 thinking that the id is invalid. I could try to create a new clean object for the server in the reducer right before calling axios. wdyt ?

Comment (deepu105) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: Can we check why it creates the empty id id field in the first place? May be we can avoidnit from happening unless a relationship is selected

Comment (agaspardcilia) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs]: I think it's caused by AvForm.

**Comment by deepu105** [comment]
Ok lets do it in the reducer for now, make a copy of the object by copying
only valid props and use a generic util for it instead of creating a method
for all reducers. May be you could use a lodash method like 'pick' for this

**Pull request merged**